File: leetcode/0118-pascals-triangle/solution.cpp
#include <vector>
using namespace std;

class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> result;
        result.push_back({1});

        for (int i = 1; i < numRows; i++) {
            const vector<int>& prevRow = result[i - 1];
            vector<int> row;
            row.push_back(1);
            for (int j = 1; j < i; j++) {
                row.push_back(prevRow[j - 1] + prevRow[j]);
            }
            row.push_back(1);
            result.push_back(row);
        }

        return result;
    }
};

File: leetcode/0118-pascals-triangle/solution.py
from typing import List


class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        result = [[1]]

        for i in range(1, numRows):
            prev_row = result[-1]
            row = [1]
            for j in range(1, i):
                row.append(prev_row[j - 1] + prev_row[j])
            row.append(1)
            result.append(row)

        return result
